hydromt/components/grid.py

```diff
@@ -36,17 +34,21 @@
 __all__ = ["GridComponent"]
 
 
-class GridComponent(ModelComponent):
+class GridComponent(SpatialModelComponent):
     """ModelComponent class for grid components.
 
     This class is used for setting, creating, writing, and reading regular grid data for a
     HydroMT model. The grid component data stored in the ``data`` property of this class is of the
     hydromt.gis.raster.RasterDataset type which is an extension of xarray.Dataset for regular grid.
     """
 
+    DEFAULT_FILENAME = "grid/grid.nc"
```

Comment:
What would be the advantage of one over the other? I have also seen some that we prefixed with an underscore.

I chose for this because I can simply call it `DEFAULT_FILENAME` and any inheriting class can access it via `GridComponent.DEFAULT_FILENAME`.

We can also consider making it a property, so that inheriting classes (`MySpecialGridComponent`) can overwrite it as follows:

```py
class GridComponent
  @property
  def default_filename(self):
    return "grid/grid.nc"

  def __init__(self, filename: Optional[str] = None):
    self.__filename = filename or self.default_filename()

class MySpecialGridComponent(GridComponent):
  @property
  def default_filename(self):
    return "some_other_place/special.nc"
```

Reply:
@savente93 What do you think? How should we handle `_DEFAULT_FILENAME` or `cls.DEFAULT_FILENAME` or `self.default_filename()`?

Reply:
I think we already have a lot of properties so I'd like to not add more unless really necessary. I don't think there is much bennefit of one over the other, but I agree that we should be consistent. I think the class variable is slightly better in the case of inheritance as Marnix mentioned so unless others have reasons not to, let's go with that. If we do, we should also bring the other components in line